rust/azure_iot_operations_connector/src/base_connector/managed_azure_device_registry.rs

```diff
@@ -1485,23 +1762,228 @@ pub enum MessageSchemaError {
     AzureDeviceRegistryError(#[from] azure_device_registry::Error),
 }
 
-type DataOperationUpdateNotification = (
-    DataOperationDefinition,                     // new data operation definition
-    Vec<Arc<destination_endpoint::Destination>>, // new default data operation destinations
-    watch::Receiver<()>, // watch receiver for when the update notification should be released to the application
-);
+type DataOperationUpdateNotification = (
+    DataOperationDefinition,                     // new data operation definition
+    Vec<Arc<destination_endpoint::Destination>>, // new default data operation destinations
+    watch::Receiver<()>, // watch receiver for when the update notification should be released to the application
+);
+
+/// Notifications that can be received for a Data Operation
+pub enum DataOperationNotification {
+    /// Indicates that the Data Operation's definition has been updated in place
+    Updated,
+    /// Indicates that the Data Operation has been deleted.
+    Deleted,
+    /// Indicates that the Data Operation received an update, but the update was not valid.
+    /// The definition is still updated in place, but the [`DataOperationClient`] should not be used until
+    /// there is a new update, otherwise the out of date definition will be used for
+    /// sending data to the destination.
+    UpdatedInvalid,
+}
+
+/// Result of a schema modification attempt
+pub enum SchemaModifyResult {
+    /// Indicates that the schema was reported successfully and status was modified
+    Reported(MessageSchemaReference),
+    /// Indicates that the schema or status were not modified
+    NotModified,
+}
+
+/// A cloneable status reporter for Data Operation status reporting.
+///
+/// This provides a way to report Data Operation status changes from outside the [`DataOperationClient`].
+#[derive(Debug, Clone)]
+pub struct DataOperationStatusReporter {
+    connector_context: Arc<ConnectorContext>,
+    asset_status: Arc<tokio::sync::RwLock<adr_models::AssetStatus>>,
+    asset_specification: Arc<std::sync::RwLock<AssetSpecification>>,
+    data_operation_ref: DataOperationRef,
+    asset_ref: AssetRef,
+}
+
```

Comment:
It strikes me that perhaps some of this should start to move to a different module. Is this *really* part of the the "managed ADR" anymore?

Reply:
Val raised a similar issue, eventually she plans on separating this based on the client (DataOperationClient + DataOperationStatusReporter in one module, etc.)